File: Backend/API/Models/Restaurants.Models/Data/RestaurantPlan.cs
using System.Collections.Generic;

namespace Models.Restaurants.Models.Data
{
    public class RestaurantPlan
    {
        public int Id { get; set; }
        public string WebSvg { get; set; }

        public Restaurant Restaurant { get; set; }
        public int RestaurantId { get; set; }
        public ICollection<PlanWall> Walls { get; set; }
        public ICollection<PlanTable> Tables { get; set; }
    }
}

File: reservation_app/lib/Models/restaurants/plan/restaurant_plan.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:reservation_app/Models/interfaces/json_format.dart';
import 'package:reservation_app/Models/restaurants/plan/plan_table.dart';
import 'package:reservation_app/Models/restaurants/plan/plan_wall.dart';

part 'restaurant_plan.g.dart';

@JsonSerializable(explicitToJson: true)
class RestaurantPlan implements JsonFormat {
  RestaurantPlan(this.id, this.webSvg, this.walls, this.tables);

  final int id;
  final String webSvg;
  final List<PlanWall> walls;
  final List<PlanTable> tables;

  factory RestaurantPlan.fromJson(Map<String, dynamic> json) => _$RestaurantPlanFromJson(json);

  Map<String, dynamic> toJson() => _$RestaurantPlanToJson(this);
}
